docs/modules/ROOT/pages/how-tos/crossplane/working_with_functions.adoc

```diff
@@ -0,0 +1,197 @@
+
+* observed: this one describes what is current state of our resources, it's read-only, all changes will be discarded
+* desired: this one is state we want to have and it's place where we can add aditional resources or manage already existing ones
+* results: this one is a logger, we can create log entries with 3 levels of severity: Normal, Warning and Fatal - Fatal breaks function's code execution
+
+
+== Preparation step 1
+To start working with Functions we must enable alfa features in Crossplane (at least in version v1.11)
+```
+helm install crossplane --namespace crossplane-system crossplane-stable/crossplane \
+    --create-namespace \
+    --set "args={--debug,--enable-composition-functions}" \
+    --set "xfn.enabled=true" \
+    --set "xfn.args={--debug}"
+```
+
+
+== Preparation step 2
+Next step is to declare our Function in our managed resource (Composition).
+It's highly recommended to use other container registry than hub.docker.io especially for development because their pull limit can block Your dev environment
+```
+spec:
+  compositeTypeRef:
+    apiVersion: vshn.appcat.vshn.io/v1
+    kind: XVSHNPostgreSQL
+  functions:
+    - name: fnio
+      type: Container
+      container:
+        image: ghcr.io/wejdross/fnio
+        imagePullPolicy: Always
+```
+
+
+== Example function 1 - Golang
+
+As a result of code below there will be ConfigMap created even though it wasn't declared in original composition. This code simply takes input from stdin and unmarshal it into xfnv1alpha1.FunctionIO{}, further we append current list of desired resources by our new resource and print it to stdout. You can see that we also create log entry which will print whole aboject we received - be carefull with that, as in my case it was almost 5000 lines. 
+
+```
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	xfnv1alpha1 "github.com/crossplane/crossplane/apis/apiextensions/fn/io/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+func main() {
+
+	funcIO := xfnv1alpha1.FunctionIO{}
+
+	x, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		log.Fatal(err)
+	}
+    // it's necessary to use sigs.k8s.io/yaml parser, otherwise it's impossible to force functions to work
+	err = yaml.Unmarshal(x, &funcIO)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	object := `
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: Object
+metadata:
+  name: testingconfigmaps
+spec:
+  providerConfigRef:
+    name: kubernetes
+  forProvider:
+    manifest:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: url-config
+        namespace: default
+      data:
+        fullURL: "https://google.pl"`
+
+	k8sapproved, err := yaml.YAMLToJSON([]byte(object))
+	if err != nil {
+		log.Fatal("from k8sapproved", err)
+	}
+
+	funcIO.Desired.Composite.Resource.Raw = funcIO.Observed.Composite.Resource.Raw
```

Comment:
I think this is unnecessary. From documentation it seems that all resources are already present in the desired state. Instead of having this line of code I would add a comment stating that the desired resource already matches from the observe.